main.go

```diff
@@ -402,11 +403,13 @@ func setupChecks(mgr ctrlmgr.Manager) {
 func isCRDDeployed(mgr ctrlmgr.Manager, gvr schema.GroupVersionResource) (bool, error) {
 	_, err := mgr.GetRESTMapper().KindFor(gvr)
 	if err != nil {
-		discoveryErr, ok := errors.Unwrap(err).(*discovery.ErrGroupDiscoveryFailed)
+		var discoveryErr *apiutil.ErrResourceDiscoveryFailed
```

Comment:
Q: We have/had some sort of confirmation via integration / e2e tests that the discovery still works, right?

Reply:
Yeah - without this change the integration test on startup. We also have integration tests that run supervisor API types.

Reply:
I don't believe we have any e2e tests for this.

Reply:
I meant more like: integration tests cover the supersvisor code path and e2e tests the other one

Sounds good!